docs/docs/tutorial/part-6/index.mdx

```diff
@@ -0,0 +1,615 @@
+    ```js:title=src/pages/blog/{mdx.slug}.js
+    import * as React from 'react'
+    import { graphql } from 'gatsby'
+    import { MDXRenderer } from 'gatsby-plugin-mdx' // highlight-line
+    import Layout from '../../components/layout'
+
+    const BlogPost = ({ data }) => { // highlight-line
+      return (
+        {/* highlight-start */}
+        <Layout pageTitle={data.mdx.frontmatter.title}>
+          <p>{data.mdx.frontmatter.date}</p>
+          <MDXRenderer>
+            {data.mdx.body}
+          </MDXRenderer>
+        {/* highlight-end */}
+        </Layout>
+      )
+    }
+
+    export const query = graphql`
+      query MyQuery($id: String) {
+        mdx(id: {eq: $id}) {
+          frontmatter {
+            title
+            date(formatString: "MMMM D, YYYY")
+          }
+          body
+        }
+      }
+    `
+
+    export default BlogPost
+    ```
+
+4. In your web browser, go to one of your blog post pages (like `localhost:8000/blog/my-first-post`). You should see the contents of your blog post rendered in their own page!
+    * Try checking the routes for your other posts, to make sure all your pages are rendering the post contents correctly.
+
+![A screenshot of the "my-first-post" blog page in a web browser. The post's title, date, and body contents are rendered.](./my-first-post.png)
+
+## Update Blog page to link to each post
+
+So far, you've used the File System Route API and GraphQL query variables to create separate pages for each of your blog posts.
+
+The last step of Part 6 is to clean up your Blog page. Instead of rendering the full contents of your blog posts, the Blog page should link out to the new post pages you just created.
+
+1. In your `src/pages/blog/index.js` file, remove the `body` field from your JSX and your page query. You can also get rid of the `MDXRenderer` component.
+    ```js:title=src/pages/blog/index.js
+    import * as React from 'react'
+    import { graphql } from 'gatsby'
+    import Layout from '../../components/layout'
+
+    const BlogPage = ({ data }) => {
+      return (
+        <Layout pageTitle="My Blog Posts">
+          {
+            data.allMdx.nodes.map(node => (
+              <article key={node.id}>
+                <h2>{node.frontmatter.title}</h2>
+                <p>Posted: {node.frontmatter.date}</p>
+              </article>
+            ))
+          }
+        </Layout>
+      )
+    }
+    
+    export const query = graphql`
+      query {
+        allMdx(sort: {fields: frontmatter___date, order: DESC}) {
+          nodes {
+            frontmatter {
+              date(formatString: "MMMM D, YYYY")
+              title
+            }
+            id
+          }
+        }
+      }
+    `
+
+    export default BlogPage
+    ```
+
+2. Add the `slug` field to your page query, and use it to turn the post title into a link to the post page.
```

Comment:
This part should teach the `gatsbyPath` API: https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/#routing-and-linking

While `slug` might directly work in this instance when people use fields that have not already "slugified" URLs like titles then the linking won't work as FS Routes creates a different path then people will link to.

I didn't want to directly change it as we might link to an example that also needs this update + it's not blocking for the usage of `slug` but we should definitely change this.

Reply:
We could mention the `gatsbyPath` API in something like a callout, couldn’t we? Either way, that seems like it shouldn’t block this PR

Reply:
I'd like to use it directly as it's the only correct way for people to get the final URL when they use the File System Route API. Any attempt of people using maybe using another third-party library to slugify the path might result in problems, too, as the library options might differ from what we're using.

It's not blocking for the whole PR as `slug` is already slugified, but that's really the exception. Normally people will use stuff that isn't already slugified.

Reply:
@LekoArts I tried using `gatsbyPath` yesterday but couldn't figure out how to get it to work.

When I test it out in GraphiQL, this query:

```graphql
query MyQuery {
  mdx(slug: {eq: "another-post"}) {
    frontmatter {
      title
    }
    gatsbyPath(filePath: "/blog/{mdx.slug}")
  }
}
```

...gives me back this response:

```json
{
  "data": {
    "mdx": {
      "frontmatter": {
        "title": "Another Post",
      },
      "gatsbyPath": "/blog/{mdx/"
    }
  },
  "extensions": {}
}
```

Could you help me figure out what I'm doing wrong?

Reply:
Given the time constraints of the livestream later today, I'm going to merge this as-is for now, and then I can push an update this afternoon when I can figure out how to get `gatsbyPath` working.

Reply:
it‘s this issue: https://github.com/gatsbyjs/gatsby/issues/30429

You could use mdx.title instead